apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/newchat/NewChatView.kt

```diff
@@ -192,6 +207,254 @@ private fun RetryButton(onClick: () -> Unit) {
+  image: @Composable () -> Unit,
+  onInfo: (@Composable () -> Unit)? = null
+) {
+  Row(
+    Modifier
+      .fillMaxWidth()
+      .sizeIn(minHeight = DEFAULT_MIN_SECTION_ITEM_HEIGHT)
+      .then(if (disabled) Modifier else Modifier.clickable(onClick = onSelected))
+      .padding(horizontal = DEFAULT_PADDING),
+    horizontalArrangement = Arrangement.SpaceBetween,
+    verticalAlignment = Alignment.CenterVertically
+  ) {
+    Row(verticalAlignment = Alignment.CenterVertically) {
+      image()
+      TextIconSpaced(false)
+      Text(title, modifier = Modifier.align(Alignment.CenterVertically))
+      if (onInfo != null) {
+        Spacer(Modifier.padding(6.dp))
+        Column(Modifier
+          .size(48.dp)
+          .clip(CircleShape)
+          .clickable(
+            enabled = !disabled,
+            onClick = { ModalManager.start.showModal { IncognitoView() } }
+          ),
+          horizontalAlignment = Alignment.CenterHorizontally,
+          verticalArrangement = Arrangement.Center
+        ) {
+          Icon(
+            painterResource(MR.images.ic_info),
+            stringResource(MR.strings.incognito),
+            tint = MaterialTheme.colors.primary
+          )
+        }
+      }
+    }
+    if (selected) {
+      Icon(painterResource(
+        MR.images.ic_check),
+        title,
+        Modifier.size(20.dp),
+        tint =  MaterialTheme.colors.primary,
+      )
+    }
+  }
+  Divider(
+    Modifier.padding(
+      start = DEFAULT_PADDING_HALF,
+      top = 4.dp,
+      end = DEFAULT_PADDING_HALF,
+      bottom = 4.dp
+    )
+  )
+}
+
+private fun filteredProfiles(users: List<User>, searchTextOrPassword: String): List<User> {
+  val s = searchTextOrPassword.trim()
+  val lower = s.lowercase()
+  return users.filter { u ->
+    if ((u.activeUser || !u.hidden) && (s == "" || u.anyNameContains(lower))) {
+      true
+    } else {
+      correctPassword(u, s)
+    }
+  }
+}
+
+@Composable
+private fun ActiveProfilePicker(
+  search: MutableState<String>,
+  contactConnection: PendingContactConnection?,
+  close: () -> Unit,
+  rhId: Long?
+) {
+  val switchingProfile = remember { mutableStateOf(false) }
+  val incognito = remember {
+    chatModel.showingInvitation.value?.conn?.incognito ?: controller.appPrefs.incognito.get()
+  }
+  val selectedProfile by remember { derivedStateOf { chatModel.currentUser.value } }
+  val searchTextOrPassword = rememberSaveable { search }
+  val profiles = remember {
+    chatModel.users.map { it.user }.sortedBy { !it.activeUser }
+  }
+  val filteredProfiles by remember {
+    derivedStateOf { filteredProfiles(profiles, searchTextOrPassword.value) }
+  }
+
+  var progressByTimeout by rememberSaveable { mutableStateOf(false) }
+
+  LaunchedEffect(switchingProfile.value) {
+    progressByTimeout = if (switchingProfile.value) {
+      delay(500)
+      switchingProfile.value
+    } else {
+      false
+    }
+  }
+
+  @Composable
+  fun profilePickerUserOption(user: User) {
```

Comment:
`@Composables` should always have Capitalized function names. And please don't include functions inside functions. It's a bad practice in Kotlin and should be avoided at any costs except if you need to pass millions of arguments to such function and you prefer to get those arguments from outer function (it makes code unreadable). If it is the case, keep your version

Reply:
yes that is the case here, would require 6 arguments to be passed around